lib/dm-core/model.rb

```diff
@@ -380,7 +380,11 @@ def first(*args)
       if limit_specified
         all(query)
       else
-        query.repository.read(query).first
+        begin
+          query.repository.read(query).first
+        rescue Query::InvalidConditionsError
+          nil
```

Comment:
Required because the checks for invalid query do not actually represent an invalid query.

One such example: `Article.first(id: nil)` Should immediately return nil, nobody expects this to raise an error.

This checks validity by asking the property `Article.properties[:id].valid?(nil)`. Obviously this is invalid (given this property is Serial/key).